tests/test_tokenization_common.py

```diff
@@ -166,15 +166,27 @@ def get_clean_sequence(self, tokenizer, with_prefix_space=False, max_length=20,
 
     def get_tokenizers(self, fast=True, **kwargs) -> List[PreTrainedTokenizerBase]:
         if fast and self.test_rust_tokenizer:
-            return [self.get_tokenizer(**kwargs), self.get_rust_tokenizer(**kwargs)]
-        return [self.get_tokenizer(**kwargs)]
+            if self.tokenizer_class is not None:
+                return [self.get_tokenizer(**kwargs), self.get_rust_tokenizer(**kwargs)]
+            else:
+                return [self.get_rust_tokenizer(**kwargs)]
+        elif self.tokenizer_class is not None:
+            return [self.get_tokenizer(**kwargs)]
+        else:
+            return []
 
     def get_tokenizer(self, **kwargs) -> PreTrainedTokenizer:
         return self.tokenizer_class.from_pretrained(self.tmpdirname, **kwargs)
 
     def get_rust_tokenizer(self, **kwargs) -> PreTrainedTokenizerFast:
         return self.rust_tokenizer_class.from_pretrained(self.tmpdirname, **kwargs)
 
+    def get_main_tokenizer(self, **kwargs) -> Union[PreTrainedTokenizer, PreTrainedTokenizerFast]:
+        if self.tokenizer_class is not None:
+            return self.get_tokenizer(**kwargs)
+        else:
+            return self.get_rust_tokenizer(**kwargs)
```

Comment:
I agree with you completely on that.  Does it fit you if: 
- I change all the tests that were only testing `get_tokenizer` (cf the following comments) to testing all tokenizers in `get_tokenizers` ?
- I delete this new method because it would not be used anymore ?